### PR TITLE
Render three color icons by blending two grayscale

### LINKED ISSUE #2558: each icon (for IconDialog) takes 0.5K of flash, we need to either figure out how to represent three-color image more efficiently than 16bpp, or draw them procedurally

### PR DESCRIPTION
**Pull request by TychoVrahe**
Saves some flash space and is a bit faster.

Currently the `BlendedImage` component draw both icons centered, if we need something else in the future it should be easy to modify.

Uses DMA2D for blending.

When DMA2D is not available, defaults to drawing one icon first and then another, which works fine but is prone to flickering, is slower and is limited to non-overlapping borders (= only works correctly if the FG icon is rendered over only one color of the BG icon).

Closes #2558

### CODE REVIEW

**Review by mmilata — COMMENTED** [review]
Got some suggestions but otherwise looking good.

fixtures.json needs updating since there are some invisible differences. I can do it for you since it's a bit tricky for UI2.

File: core/embed/rust/src/ui/model_tt/theme.rs [hunk 1]
```diff
@@ -55,11 +55,17 @@ pub const ICON_WARN: &[u8] = include_res!("model_tt/res/warn-icon.toif");
...
-// Large, color icons.
-pub const IMAGE_WARN: &[u8] = include_res!("model_tt/res/warn.toif");
-pub const IMAGE_SUCCESS: &[u8] = include_res!("model_tt/res/success.toif");
-pub const IMAGE_ERROR: &[u8] = include_res!("model_tt/res/error.toif");
-pub const IMAGE_INFO: &[u8] = include_res!("model_tt/res/info.toif");
+// Large, three-color icons.
+pub const WARN_COLOR: Color = Color::rgb(0xC1, 0x90, 0x09);
+pub const INFO_COLOR: Color = Color::rgb(0x00, 0x56, 0xBE);
+pub const SUCCESS_COLOR: Color = Color::rgb(0x39, 0xA8, 0x14);
+pub const ERROR_COLOR: Color = Color::rgb(0xCD, 0x49, 0x49);
```
[the remark this answers is not in the document]
Comment (TychoVrahe) [reply core/embed/rust/src/ui/model_tt/theme.rs]: fixed, including the change to hex: https://github.com/trezor/trezor-firmware/pull/2573/commits/e7de7a60a65daee701d13ab97b9fdbc779b7e9ce

File: core/embed/rust/src/ui/component/image.rs [hunk 2]
```diff
@@ -40,3 +41,110 @@ impl Component for Image {
...
+
+impl Component for BlendedImage {
+    type Msg = Never;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        self.area = bounds;
+
+        if let Some((bg_size, _)) = toif_info(self.bg) {
+            self.bg_offset =
+                (self.area.center() - Offset::new(bg_size.x / 2, bg_size.y / 2)).into();
```
[the remark this answers is not in the document]
Comment (TychoVrahe) [reply core/embed/rust/src/ui/component/image.rs]: changed, inc. removal of the `self.area`  https://github.com/trezor/trezor-firmware/pull/2573/commits/6aec7f839c02754842337f356eff9ec1a50e277e

**Comment by TychoVrahe** [comment]
regarding the fixtures, i don't know how to but i would like to learn - unless that process will change for UI2 after we switch, in that case i would prefer to learn afterwards :)

**Comment by mmilata** [comment]
Normally you'd either run `make test_emu_ui_record` or run (possibly on CI) the UI test and apply `fixtures.suggestion.json`.

On UI2 `test_emu_ui_record` doesn't work because some of the tests fail (or hang without PYTEST_TIMEOUT), also the `core device ui2 test` is manual so the fixtures sometimes go out of sync with master and master diff shows more differences than what your branch causes. After we switch to UI2 this should be resolved, until then feel free to let me know and I'll do it for you.

**Review by mmilata — APPROVED** [review]
LGTM, I'll add the fixtures, squash, and merge it if that's OK.

**Comment by mmilata** [comment]
UI diff https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/3212080139/artifacts/master_diff/index.html

There are small differences in the edges but to my untrained eye the images are similar enough to ignore.

icon3

**Pull request merged**